### PR TITLE
Improve inactive group view

### PR DESCRIPTION
**Pull request by cascheberg**
### Contributor checklist

- [X] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [X] I have tested my contribution
- [X] My contribution is fully baked and ready to be merged as is
- [X] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

---
### Description

This PR improves **inactive group** and **blocked contact** conversation views:
* merge `initializeEnabledCheck()` and `setBlockedUserState()` into new `updateConversationState()`
* hide "disappearing messages" menu item
* hide call button
* hide "reset secure session"
* hide "invite" menu item
* hide compose panel in inactive groups
* rearrange menu items so that the first five and the last item are consistent for single and group conversations.

**Screenshot before** (single | active group | inactive group | blocked single)
before

**Screenshot after** (single | active group | inactive group | blocked single)
after

// FREEBIE

### CODE REVIEW

**Comment by scienmind** [comment]
Is there a reason to leave "Conversation Settings" and "Mute Notifications" for inactive groups?

(My understanding was that once you left the group, you won't receive anything there and hence muting/customizing sounds, etc. is not relevant there)

**Comment by cascheberg** [comment]
@scienmind The one thing that I could think of was that someone repeatedly re-adds you to the group, so that you might want to mute it.

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 1]
```diff
@@ -1160,6 +1158,8 @@ private void setActionBarColor(MaterialColor color) {
   }
 
   private void setBlockedUserState(Recipients recipients) {
+    if (isGroupConversation()) return;
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: why was this necessary?

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 2]
```diff
@@ -1210,8 +1210,7 @@ private boolean isSelfConversation() {
   }
 
   private boolean isGroupConversation() {
-    return getRecipients() != null &&
-        (!getRecipients().isSingleRecipient() || getRecipients().isGroupRecipient());
+    return getRecipients() != null && !isSingleConversation();
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: why is this necessary?

**Comment by moxie0** [comment]
looks nice

File: src/org/thoughtcrime/securesms/ConversationActivity.java (hunk 2 above)
Comment (cascheberg) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: This is to make clear how `isGroupConversation` behaves in relation to `isSingleConversation`, i.e. only one of them is true (which is different from how `isSingleRecipient` and `isGroupRecipient` work).

File: src/org/thoughtcrime/securesms/ConversationActivity.java (hunk 1 above)
Comment (cascheberg) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: `setBlockedUserState` is setting `composePanel.setVisibility`, which now conflicts with `initializeEnabledCheck`. This was sufficient to resolve the problem.

Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: Except now the method is no longer idempotent. Why don't you start with just the menu changes.

Comment (cascheberg) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: I can do that. But I'd be glad if you proposed a different approach.

**Comment by cascheberg** [comment]
Rebased and rewritten PR which checks for inactive groups and blocked contacts:
* merge `initializeEnabledCheck()` and `setBlockedUserState()` into new `updateConversationState()`
* hide "disappearing messages" menu item
* hide "reset secure session"
* hide "invite" menu item
* hide "add attachment" menu item

**Comment by cascheberg** [comment]
Rebased and slightly improved with regard to #6400

**Comment by cascheberg** [comment]
PR rebased, reduced amount of required changes, and tested again.

**Comment by cascheberg** [comment]
PR rebased, tested and screenshots updated.

**Screenshot before** (single | active group | inactive group | blocked single)
before

**Screenshot after** (single | active group | inactive group | blocked single)
after

**Comment by cascheberg** [comment]
PR is outdated and issue has been fixed for the most part
